ESSArch_Core/maintenance/views.py

```diff
@@ -87,30 +87,20 @@ def run(self, request, pk=None):
     def preview(self, request, pk=None):
         job = self.get_object()
         ips = job.rule.information_packages.filter(appraisal_date__lte=timezone.now(), active=True)
-        found = []
+        found_files = []
 
         for ip in ips:
             datadir = os.path.join(ip.policy.cache_storage.value, ip.object_identifier_value)
             if job.rule.specification:
                 for pattern in job.rule.specification:
-                    for path in iglob(datadir + '/' + pattern):
-                        if os.path.isdir(path):
-                            for root, dirs, files in walk(path):
-                                rel = os.path.relpath(root, datadir)
-
-                                for f in files:
-                                    found.append({'ip': ip.object_identifier_value, 'document': os.path.join(rel, f)})
-
-                        elif os.path.isfile(path):
-                            rel = os.path.relpath(path, datadir)
-                            found.append({'ip': ip.object_identifier_value, 'document': rel})
+                    found_files = find_all_files(datadir, ip, pattern)
```

Comment:
The list must be updated, not replaced, on each iteration of the rule specification

```suggestion
                    found_files.extend(find_all_files(datadir, ip, pattern))
```